tox.ini

```diff
@@ -0,0 +1,10 @@
+[tox]
+envlist = py{37,38,39,310}
+isolated_build = true
+
+[testenv]
+allowlist_externals = poetry
+commands_pre =
+    poetry install --no-root --only main --sync
+commands =
+    coverage run -m unittest discover tests/
```

Comment:
This just adds testing facilities ?

Reply:
yes!  see CONTRIBUTING.md for how to test with tox. 

It basically installs the package with different python versions, install all dependencies listed in the pyproject.toml file, and runs pytest on the existing tests. It's aimed to test the package with different versions of python. 

You'll need to provide access to different versions of python though, it will skip them if not found. I use pyenv to manage different python versions.